backend/program/symlink.py

```diff
@@ -27,44 +28,38 @@ class Symlinker():
         host_path (str): The absolute path of the host mount.
         symlink_path (str): The path where the symlinks will be created.
     """
-    def __init__(self, _):
+    def __init__(self, *_):
         self.key = "symlink"
         self.settings = SymlinkConfig(**settings.get(self.key))
         self.initialized = self.validate()
         if not self.initialized:
-            logger.error("Symlink initialization failed due to invalid configuration.")
-            return
+            raise ValueError("Symlink initialization failed due to invalid configuration.")
         logger.info("Rclone path symlinks are pointed to: %s", self.settings.host_path)
         logger.info("Symlinks will be placed in: %s", self.library_path)
         logger.info("Symlink initialized!")
         self.initialized = True
 
     def validate(self):
         """Validate paths and create the initial folders."""
-        host_path = Path(self.settings.host_path) if self.settings.host_path else None
-        container_path = Path(self.settings.container_path) if self.settings.container_path else None
-        if not host_path or not container_path or host_path == Path('.') or container_path == Path('.'):
-            logger.error("Host or container path not provided, is empty, or is set to the current directory.")
-            return False
-        if not host_path.is_absolute():
-            logger.error(f"Host path is not an absolute path: {host_path}")
-            return False
-        if not container_path.is_absolute():
-            logger.error(f"Container path is not an absolute path: {container_path}")
-            return False
-        try:
-            if not host_path.is_dir():
-                logger.error(f"Host path is not a directory or does not exist: {host_path}")
+        for name, path in self.settings.__iter__():
+            if name == "symlink_path" and path is None:
+                continue
+            elif path == Path('.'):
+                logger.error(f"{name} is set to the current directory.")
                 return False
-            # if not container_path.is_dir():
-            #     logger.error(f"Container path is not a directory or does not exist: {container_path}")
-            #     return False
-            if Path(self.settings.host_path / "__all__").exists() and Path(self.settings.host_path / "__all__").is_dir():
+            elif not path.is_absolute():
+                logger.error(f"{name} is not an absolute path: {path}")
+                return False
+            elif not path.is_dir():

```

Comment:
Not sure why we wouldn't want to validate all entries as dirs

Reply:
I felt the same way as well, but Mini was having issues. Need to gather more info from @AyushSehrawat on what kind of issues exactly and **how** it was working for him before, if it even was? Was it playing on plex from the symlink path Mini? Does `/mount` exist? 

It's also possible that I need to understand how `os.symlink()` works and if it requires the path to be a valid path to create a symlink in the OS. This is what I'm thinking happened, is that it doesnt need a valid path to create the link.. but since he plays the symlinks in Plex, that path is valid in plex which works as intended for the `container_path` setting.